### PR TITLE
CPU/DFT Info Updates

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
Moves the general info to the `info` folder where these items are effectively singletons (see tests for usage examples). These are also not imported until requested due to the potential size of their data objects.

The CPU search is a bit strange and requires some special cases. It will certainly need continuous updates and overhauls. I will add docs for this after @mattwelborn gives this a go in a project.

## Changelog description
Updates the DFT Info objects and adds a info stub of CPU information.

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcelemental/info/cpu_info.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+from enum import Enum
+from typing import Dict, Optional
+
+from pydantic import Field
+
+from ..models import ProtoModel
+
+
+class VendorEnum(str, Enum):
+    """Allowed quantum chemistry driver values.
```
Comment (mattwelborn) [inline qcelemental/info/cpu_info.py]: ??

File: qcelemental/info/cpu_info.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        self.name = context
+
+    def __str__(self) -> str:
+        return "ProcessorContext(context='{}')".format(self.name)
+
+    def process_names(self, name):
+        name = name.lower()
+        name = name.replace("(tm)", "").replace("(r)", "").replace("™", "")
+        name = name.replace("  ", " ").replace("\t", " ")
```
Comment (mattwelborn) [inline qcelemental/info/cpu_info.py]: `name = ' '.join(name.split())`

File: raw_data/cpu_data/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Fileame: intel_cpu_database.json
+Download date: Jan 7, 2020
+Link: https://github.com/divinity76/intel-cpu-database
+
+Verified data from ark.intel.com from the following repository
+
+Fileame: Intel_ARK_ComparisonChart_2020_01_07.csv
+Download date: Jan 7, 2020
+Link: https://ark.intel.com/content/www/us/en/ark/compare.html?productIds=194146,194145,195437,195434,192473,192472,192478,192471,192475,192470,192482,192481,192480,192477,192476,192474,192467,192465,193972,192451,193951,192447,192446,193969,192442,192448,193960,193959,192439,193955,193958,193954,193952,192450,193962,192445,193970,193950,193388,193953,193397,193564,193396,193392,193393,193391,193394,193389,193553,193385,193384,193382,193390,193381,193949,193957,192437,192444,192443,192440
```
Comment (mattwelborn) [inline raw_data/cpu_data/README.md]: It would be nice if we could somehow get a list of all productIds

Comment (mattwelborn) [reply raw_data/cpu_data/README.md]: Hundreds of results here.

https://ark.intel.com/content/www/us/en/ark/search/featurefilter.html?productType=873&2_StatusCodeText=4

Comment (mattwelborn) [reply raw_data/cpu_data/README.md]: Even better:
https://ark.intel.com/libs/apps/intel/support/ark/advancedFilterSearch?productType=873&2_StatusCodeText=4&forwardPath=/content/www/us/en/ark/search/featurefilter.html&pageNo=25&sort=&sortType=

Edit pageNo to get 25 pages worth of productIDs

Comment (dgasmith) [reply raw_data/cpu_data/README.md]: Can we split this out as an issue?

File: qcelemental/info/cpu_info.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+class ProcessorContext:
+    """Information about Process information.
+
+    Parameters
+    ----------
+    context : {'default'}
+        Origin of loaded data.
+
+    Attributes
+    ----------
```
Comment (mattwelborn) [inline qcelemental/info/cpu_info.py]: Fix docstring

File: qcelemental/info/dft_info.py [hunk 5]
```diff
@@ -22,38 +22,45 @@ class DFTFunctional(ProtoModel):
...
 
 
 # singleton
-dftfunctionalinfo = DFTFunctionalInfo("PSI4")
+dftfunctionalinfo = DFTFunctionalContext("default")
+
+
+def get(name: str) -> DFTFunctionalInfo:
+
+    name = name.lower().split("-d")[0].split("-nlc")[0]
```
Comment (mattwelborn) [inline qcelemental/info/dft_info.py]: Counter example: `core-dsd-blyp`

File: raw_data/cpu_data/README.md (hunk 3 above)
Comment (mattwelborn) [reply raw_data/cpu_data/README.md]: Sure, #188.

File: qcelemental/info/cpu_info.py (hunk 2 above)
Comment (dgasmith) [reply qcelemental/info/cpu_info.py]: I now use the same replace as in the `build_dft_info.py` to ensure they stay in sync.

File: qcelemental/info/dft_info.py (hunk 5 above)
Comment (dgasmith) [reply qcelemental/info/dft_info.py]: See above.

**Pull request merged**